### DISCUSSION TITLE
Using Managed Identity or Service Principal for Azure DevOps Authentication in ArgoCD

### DISCUSSION THREAD

**Discussion by ktarnaris** [argoproj/argo-cd#20999] [Q&A]
I am currently running ArgoCD v2.12.6 on an AKS cluster within Azure infrastructure. I want to avoid using a PAT (Personal Access Token) for authentication with Azure DevOps. Instead, I am exploring alternatives like Managed Identity or a Service Principal. Does anyone know if this approach is feasible?

**Comment by Jorik-VanLooy** [comment]
Hi ktarnaris,

I had the same question and i found following link.
https://stackoverflow.com/questions/67639199/is-it-possible-to-use-service-principal-to-access-git-repository-on-azure
following the link you could login via a service principal and get an access token to login to your azure devops repos.
The service principal needs the correct permissions in azure devops of course but I think it should be feasible.

**Reply by bartvanackooij** [reply]
Hi,

The approach suggested by Jorik-VanLooy can work, although the token you retrieve and can use instead of the PAT is only valid for 1 hour.

**Reply by thikade** [reply]
... which makes it a nice solution for eg pipeline tasks where you simply clone / commit once (< 1hour), but not as a credential continuously used by ArgoCD repo server    :-(
